scrapy/spidermiddlewares/urllength.py

```diff
@@ -7,72 +7,49 @@
 from __future__ import annotations
 
 import logging
-import warnings
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING
 
-from scrapy.exceptions import NotConfigured, ScrapyDeprecationWarning
-from scrapy.http import Request, Response
+from scrapy.exceptions import NotConfigured
+from scrapy.spidermiddlewares.base import BaseSpiderMiddleware
 
 if TYPE_CHECKING:
-    from collections.abc import AsyncIterable, Iterable
-
     # typing.Self requires Python 3.11
     from typing_extensions import Self
 
-    from scrapy import Spider
     from scrapy.crawler import Crawler
-    from scrapy.settings import BaseSettings
+    from scrapy.http import Request, Response
 
 
 logger = logging.getLogger(__name__)
 
 
-class UrlLengthMiddleware:
+class UrlLengthMiddleware(BaseSpiderMiddleware):
+    crawler: Crawler
+
     def __init__(self, maxlength: int):
         self.maxlength: int = maxlength
 
-    @classmethod
-    def from_settings(cls, settings: BaseSettings) -> Self:
-        warnings.warn(
-            f"{cls.__name__}.from_settings() is deprecated, use from_crawler() instead.",
```

Comment:
This is removed early but I don't think it's a big deal (we should try to mention it in the release notes though).

Reply:
Maybe edit `news.rst` already as part of this PR, so we don’t forget.